### PR TITLE
WT-3972 Allow more than 64K cursors on a single table.

### PR DESCRIPTION
**Pull request by michaelcahill**

### CODE REVIEW

File: src/support/mtx_rw.c [hunk 1]
```diff
@@ -48,9 +48,8 @@
  *		uint8_t current;		// Current ticket
  *		uint8_t next;			// Next available ticket
  *		uint8_t reader;			// Read queue ticket
- *		uint8_t __notused;		// Padding
- *		uint16_t readers_active;	// Count of active readers
- *		uint16_t readers_queued;	// Count of queued readers
+ *		uint8_t readers_queued;		// Count of queued readers
```
Comment (agorrod) [inline src/support/mtx_rw.c]: With this change, I think the comment [here](https://github.com/wiredtiger/wiredtiger/blob/develop/src/support/mtx_rw.c#L209) should be brought up to the general description block - I presume that's the reason why it's OK to use a smaller number for queued readers?

Comment (michaelcahill) [reply src/support/mtx_rw.c]: Agreed, that comment is the reason this change is possible.  I'll update with your suggestion.

File: test/suite/test_cursor14.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        ('table-r-complex', dict(type='table:', keyfmt='r',
+            dataset=ComplexDataSet)),
+        ('table-S-complex', dict(type='table:', keyfmt='S',
+            dataset=ComplexDataSet)),
+        ('table-S-complex-lsm', dict(type='table:', keyfmt='S',
+            dataset=ComplexLSMDataSet)),
+    ])
+
+    # WT_CURSOR.insert doesn't leave the cursor positioned, verify any
+    # subsequent cursor operation fails with a "key not set" message.
```
Comment (sueloverso) [inline test/suite/test_cursor14.py]: This comment seems entirely unrelated to the code below it.

Comment (michaelcahill) [reply test/suite/test_cursor14.py]: Yes, you're right -- this was inherited from whatever test I started with.  I've removed it.

**Comment by agorrod** [comment]
lgtm
